src/Altinn.AccessManagement.Core/Services/AppsInstanceDelegationService.cs

```diff
@@ -0,0 +1,292 @@
+    {
+        ValidateAndGetSignificantResourcePartsFromResource(rights.FirstOrDefault()?.Resource, out List<UrnJsonTypeValue> firstResource, resourceid);
+        int counter = -1;
+
+        foreach (RightInternal rightV2 in rights)
+        {
+            counter++;
+
+            bool valid = ValidateAndGetSignificantResourcePartsFromResource(rightV2.Resource, out List<UrnJsonTypeValue> currentResource, resourceid);
+            if (!valid)
+            {
+                errors.Add(ValidationErrors.InvalidResource, $"Rights[{counter}]/Resource");
+                continue;
+            }
+
+            foreach (UrnJsonTypeValue urnJsonTypeValue in currentResource)
+            {
+                if (!firstResource.Contains(urnJsonTypeValue))
+                {
+                    errors.Add(ValidationErrors.InvalidResource, $"Rights[{counter}]/Resource");
+                }
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task<Result<AppsInstanceDelegationResponse>> Delegate(AppsInstanceDelegationRequest appsInstanceDelegationRequest, CancellationToken cancellationToken = default)
+    {
+        AppsInstanceDelegationResponse result = new AppsInstanceDelegationResponse();
+
+        ValidationErrorBuilder errors = default;
+
+        // Fetch from and to from partyuuid
+        (UuidType Type, Guid? Uuid) from = await TranslatePartyUuidToPersonOrganizationUuid(appsInstanceDelegationRequest.From);
+        (UuidType Type, Guid? Uuid) to = await TranslatePartyUuidToPersonOrganizationUuid(appsInstanceDelegationRequest.To);
+
+        if (from.Type == UuidType.NotSpecified)
+        {
+            errors.Add(ValidationErrors.InvalidPartyUrn, "From");
+        }
+
+        if (to.Type == UuidType.NotSpecified)
+        {
+            errors.Add(ValidationErrors.InvalidPartyUrn, "To");
+        }
+
+        // Validate Resource and instance 1. All rights include resource 2. All rights resource is identical to central value
+        AddValidationErrorsForResourceInstance(ref errors, appsInstanceDelegationRequest.Rights, appsInstanceDelegationRequest.ResourceId);
+
+        // Fetch rights valid for delegation
+        ServiceResource resource = (await _resourceRegistryClient.GetResourceList(cancellationToken)).Find(r => r.Identifier == appsInstanceDelegationRequest.ResourceId);
+        List<Right> delegableRights = null;
+
+        if (resource == null)
+        {
+            errors.Add(ValidationErrors.InvalidResource, "appInstanceDelegationRequest.Resource");
+        }
+        else
+        {
+            RightQueryForApp resourceQuery = new RightQueryForApp { OwnerApp = appsInstanceDelegationRequest.PerformedBy, Resource = resource.AuthorizationReference };
+
+            try
+            {
+                delegableRights = await _pip.GetDelegableRightsByApp(resourceQuery, cancellationToken);
```

Comment:
CanDelegate on the rights are null, and never checked after return from _pip.GetDelegableRightsByApp